lib/kafo/puppet_command.rb

```diff
@@ -40,7 +40,7 @@ def self.search_puppet_path(bin_name)
       bin_path = (::ENV['PATH'].split(File::PATH_SEPARATOR) + ['/opt/puppetlabs/bin']).find do |path|
         File.executable?(File.join(path, 'puppet')) && File.executable?(File.join(path, bin_name))
       end
-      File.join([bin_path, bin_name].compact)
+      bin_path ? File.realpath(File.join(bin_path, bin_name)) : bin_name
```

Comment:
From my testing, this will not unfortunately work. The way the Puppet AIO wrapper code works, it is not intended to be called directly. So the symlinks resolve the wrapper, but the wrapper expects that and if it is called on itself it enters an infinite loop.

For example, puppet bin:
```
/opt/puppetlabs/bin/puppet -> /opt/puppetlabs/puppet/bin/wrapper.sh
```

Which this new code will resolve to return `/opt/puppetlabs/puppet/bin/wrapper.sh` as the bin file. However, that wrapper code does the following:

```
COMMAND=`basename "${0}"`

exec /opt/puppetlabs/puppet/bin/${COMMAND} "$@"
```

Which, if you call the `wrapper.sh` directly, `COMMAND` resolves to `wrapper.sh` and it will keep calling itself endlessly.

Reply:
Ok thanks for that. Sounds like we should perform `File.realpath` and see if **that** is in `/opt/puppetlabs`, but still use the original path we found.